include/mbedtls/config_adjust_legacy_from_psa.h

```diff
@@ -440,8 +440,43 @@
 
 /* End of ECC section */
 
+#if defined(PSA_WANT_DH_RFC7919_2048)
+#if !defined(MBEDTLS_PSA_ACCEL_DH_RFC7919_2048)
+#define MBEDTLS_PSA_BUILTIN_DH_RFC7919_2048 1
+#define MBEDTLS_PSA_DH_ACCEL_INCOMPLETE_PARAMS
+#endif /* !MBEDTLS_PSA_BUILTIN_DH_RFC7919_2048 */
+#endif /* PSA_WANT_DH_RFC7919_2048 */
+
+#if defined(PSA_WANT_DH_RFC7919_3072)
+#if !defined(MBEDTLS_PSA_ACCEL_DH_RFC7919_3072)
+#define MBEDTLS_PSA_BUILTIN_DH_RFC7919_3072 1
+#define MBEDTLS_PSA_DH_ACCEL_INCOMPLETE_PARAMS
+#endif /* !MBEDTLS_PSA_BUILTIN_DH_RFC7919_3072 */
+#endif /* PSA_WANT_DH_RFC7919_3072 */
+
+#if defined(PSA_WANT_DH_RFC7919_4096)
+#if !defined(MBEDTLS_PSA_ACCEL_DH_RFC7919_4096)
+#define MBEDTLS_PSA_BUILTIN_DH_RFC7919_4096 1
+#define MBEDTLS_PSA_DH_ACCEL_INCOMPLETE_PARAMS
```

Comment:
Even though it functionally works as is, it seems a bit counterintuitiv to define `MBEDTLS_PSA_DH_ACCEL_INCOMPLETE_PARAMS` four times in a row

Should we do something like :

~~~~~
#if defined(PSA_WANT_DH_RFC7919_3072) || defined(MBEDTLS_PSA_BUILTIN_DH_RFC7919_2048)  ...
#define MBEDTLS_PSA_DH_ACCEL_INCOMPLETE_PARAMS
#endif
~~~~~

Not a requirement for reword, just a point of discussion.

Reply:
For every `PSA_WANT` we need to match the corresponding `!MBEDTLS_PSA_ACCEL` in order to define `MBEDTLS_PSA_DH_ACCEL_INCOMPLETE_PARAMS`. A similar pattern was also used in the past for `PSA_WANT_KEY_TYPE_ECC` symbols (above on the same file).

Reply:
I think if defining the same macro potentially in different places is a problem, indeed we could do
```c
#if (WANT_PARAM_A && !ACCEL_PARAM_A) || \
    (WANT_PARAM_B && !ACCEL_PARAM_B) || \
    ...
    (WANT_PARAM_X && !ACCEL_PARAM_X)
#define INCOMPLETE_PARAMS
```
instead of
```c
#if (WANT_PARAM_A && !ACCEL_PARAM_A)
#define INCOMPLETE_PARAMS
#endif
#if (WANT_PARAM_B && !ACCEL_PARAM_B)
#define INCOMPLETE_PARAMS
#endif
```

I just noticed we're actually a bit inconsistent here: as Valerio points out, [ECC is using the second style](https://github.com/Mbed-TLS/mbedtls/blob/development/include/mbedtls/config_adjust_legacy_from_psa.h#L49) but it turns out [block ciphers are using the first style](https://github.com/Mbed-TLS/mbedtls/blob/development/include/mbedtls/config_adjust_legacy_from_psa.h#L682).

Now that you mention it, the first style is indeed more compact and has the advantage of centralising the definition of the `INCOMPLETE` macro in one place, with may be more pleasant when navigating the code based on tags or such.